lib/puppet/provider/service/service.rb

```diff
@@ -36,6 +36,9 @@ def ucommand(type, fof = true)
     else
       cmd = [send("#{type}cmd")].flatten
     end
+    if (( Facter.value(:operatingsystem) == "Solaris" ) and ( type =~ /^start/ ))
```

Comment:
I'll need to do some additional investigation to confirm, but I suspect this would apply to other Solaris-derived platforms. It looks like spec tests fail if we changen```Facter.value(:operatingsystem)``` to ```Facter.value(:osfamily)```. If this is the intended design, the Solaris-derived platforms may need to be enumerated explicitly here.